smart_meter_texas/__init__.py

```diff
@@ -89,6 +92,49 @@ async def read_meter(self, client: Client):
                     _LOGGER.error("Unknown meter reading status: %s", status)
                     raise SmartMeterTexasAPIError(f"Unknown meter status: {status}")
 
+    async def get_daily(self, client: Client, ts=None):
+        """Get the daily reading of the meter"""
+
+        # Default to yesterday. Daily reading for "today" won't be available until tomorrow
+        if ts is None:
+            ts = datetime.date.today() - datetime.timedelta(days=1)
+
+        # {"esiid":"NNNNNNNNNNNNNNNNN","startDate":"07/19/2023","endDate":"07/19/2023"}
+        payload = {
+            "esiid": self.esiid,
+            "startDate": ts.strftime("%m/%d/%Y"),
+            "endDate": ts.strftime("%m/%d/%Y"),
+        }
+
+        _LOGGER.debug(f"Getting daily reading for '{ts}' using {payload}")
+        json_response = await client.request(
+            READING_DAILY,
+            json=payload,
+        )
+
+        try:
+            _LOGGER.debug(f"Daily reading data {json_response}")
+            data = json_response["dailyData"]
+            if len(data) != 1:
```

Comment:
I think this also needs to handle the instance where no data is returned

Reply:
That would be the case where `len(data) == 0` which is handled by `!= 1`

Reply:
Yes but if I try to get today for example (expecting it to fail), I get no data points, not "too many".

```
DEBUG:smart_meter_texas:Getting daily reading for '2023-08-05' using {'esiid': 'xxxxxxxxxxxx', 'startDate': '08/05/2023', 'endDate': '08/05/2023'}
DEBUG:smart_meter_texas:Daily reading data {'dailyData': []}
Traceback (most recent call last):
  File "/Users/graham/Developer/smart_meter_texas/examples/get_daily_meter_reading.py", line 44, in <module>
    loop.run_until_complete(main())
  File "/Users/graham/.pyenv/versions/3.10.3/lib/python3.10/asyncio/base_events.py", line 646, in run_until_complete
    return future.result()
  File "/Users/graham/Developer/smart_meter_texas/examples/get_daily_meter_reading.py", line 38, in main
    await meter.get_daily(client, ts=date)
  File "/Users/graham/Developer/smart_meter_texas/smart_meter_texas/__init__.py", line 119, in get_daily
    raise SmartMeterTexasAPIError("Too many data points returned")
smart_meter_texas.exceptions.SmartMeterTexasAPIError: Too many data points returned
```

Reply:
@grahamwetzler I'm not liking this solution. I think that it should return empty array. I think that allows the user better handling vs the exception throwing.

```
# 
# Get latest end of day reading. This is usually midnight of the previous day.
#
yesterday = datetime.datetime.now(tz=gettz("America/Chicago")) - datetime.timedelta(days=1)
latest_reading = await meter.get_daily(client, yesterday.date())

if len(latest_reading) == 0:
  # try again, but go back another day
  twoDaysAgo = yesterday - datetime.timedelta(days=2)
  latest_reading = await meter.get_daily(client, twoDaysAgo.date())
```

vs catching the exception, or having to nest try-catch blocks. Thoughts?

Reply:
Yep, I think returning `None` is the best option here. Maybe also raise a warning?